src/core_plugins/interpreter/public/socket.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import chrome from 'ui/chrome';
+import io from 'socket.io-client';
+import { functionsRegistry } from '@kbn/interpreter-utils/common/lib/functions_registry';
+
+const basePath = chrome.getBasePath();
+export const socket = io(undefined, { path: `${basePath}/socket.io` });
+
+socket.on('getFunctionList', () => {
+  socket.emit('functionList', functionsRegistry.toJS());
```

Comment:
we no longer need to wait for the loadBrowserPlugins, everything should be registered at this time